### ISSUE TITLE
Docker agent stop and build forever pending

### ISSUE THREAD

**Issue by cicerocomp** [drone/drone#1749]
Version 0.5. 

agent log: 
```
panic: send on closed channel

goroutine 783 [running]:
panic(0x83cf80, 0xc8201a2ce0)
       	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/drone/drone-exec/build.(*Pipeline).step.func2(0xc82072a320)
       	/go/src/github.com/drone/drone-exec/build/pipeline.go:114 +0x62
created by github.com/drone/drone-exec/build.(*Pipeline).step
       	/go/src/github.com/drone/drone-exec/build/pipeline.go:115 +0x64
```
bug happens sporadically.

**Comment by bradrydzewski** [comment]
github.com/drone/drone-exec in the logs indicates this is 0.4 and not 0.5. we are not backporting fixes to 0.4 at this time

**Comment by bradrydzewski** [comment]
Or this means u are using an old agent prototype. Please ensure you use `drone/drone agent` and not `drone/drone-agent`. 

Also per issue template let's use the gitter room to triage problems before creating github issues. Thanks!
